src/main/java/org/terasology/anatomy/AnatomySkeleton/SkeletalAuthoritySystem.java

```diff
@@ -27,48 +27,57 @@
 import org.terasology.entitySystem.systems.RegisterMode;
 import org.terasology.entitySystem.systems.RegisterSystem;
 import org.terasology.entitySystem.systems.UpdateSubscriberSystem;
+import org.terasology.logic.delay.DelayManager;
+import org.terasology.logic.delay.DelayedActionTriggeredEvent;
 import org.terasology.math.TeraMath;
 import org.terasology.registry.In;
+import org.terasology.rendering.assets.skeletalmesh.Bone;
 
 import java.util.Map;
 
 /**
  * This authority system manages the Skeletal system health updates.
  */
 @RegisterSystem(value = RegisterMode.AUTHORITY)
-public class SkeletalAuthoritySystem extends BaseComponentSystem implements UpdateSubscriberSystem {
+public class SkeletalAuthoritySystem extends BaseComponentSystem {
     @In
     private org.terasology.engine.Time time;
 
     @In
     private EntityManager entityManager;
 
+    @In
+    private DelayManager delayManager;
+
     private float bluntDamageMultiplier = 1.5f;
 
+    private String SKELETAL_REGEN_PREFIX = "Skeletal:Regen:";
+
     @Override
-    public void update(float delta) {
+    public void initialise() {
```

Comment:
Good optimization, although I wonder if this lifecycle stage might be too early in some cases. Initializing IIRC is more for the system to initialize itself in some fashion, with the overall entity base not necessarily loaded yet (but maybe enough did load for it to appear to work)

An additional optimization would be to only consider entities with *injured* bones in need of regen, which you could then tie to the event (during regular gameplay) that actually injured the bone in the first place, rather than need to do anything at all on init :-)